CsharpExtras/Map/Dictionary/Curry/CurryDictionaryBase.cs

```diff
@@ -11,6 +11,11 @@ abstract class CurryDictionaryBase<TKey, TVal> : ICurryDictionary<TKey, TVal>
         public abstract NonnegativeInteger Arity { get; }
         public abstract IEnumerable<IList<TKey>> Keyset();
 
+        public IEnumerable<(IList<TKey>, TVal)> KeyValuePairs() =>
```

Comment:
I think it is worth adding a couple of tests for these new methods (`KeyValuePairs` and `Values`). Just to make sure they return the expected keys and do not fail on empty dictionaries.